src/server/index.js

```diff
@@ -149,16 +152,18 @@ async function NextAuthHandler (req, res, userOptions) {
           return res.json({ csrfToken })
         case 'signin':
           if (pages.signIn) {
-            let signinUrl = `${pages.signIn}${pages.signIn.includes('?') ? '&' : '?'}callbackUrl=${req.options.callbackUrl}`
+            // prepend locale to custom page path if available
+            let signinUrl = `${locale ? `/${locale}` : ''}${pages.signIn}${pages.signIn.includes('?') ? '&' : '?'}callbackUrl=${req.options.callbackUrl}`
```

Comment:
What a mess with the URLs... 😅 Not your fault though, but I think we need to clean it up a bit somehow...

Have been thinking of something like this lately:

```js
class URLRelative extends URL {
  constructor (path) {
    super(new URL(path, 'http://n'))
  }

  toString () {
    return `${this.pathname}${this.search}`
  }
}
```

Then we could utilize the URL API, also for relative URLs.